docs/src/architecture/passwordless_authentication.md

```diff
@@ -0,0 +1,41 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Yubikey Passwordless Authentication
+
+This section describes Ghaf reference implementation for passwordless authentication for fast identity online (FIDO). The reference implementation has been created with Yubico Yubikeys.
+The implementation is modular and configurable - thus enabling also other implementations.
+
+This section describes Yubikey Passwordless Authentication and how to create U2F keys.
+
+The reader is expected to know the fundamentals of [FIDO](https://fidoalliance.org/specifications/), [Yubico](https://www.yubico.com/) and [U2F](https://developers.yubico.com/U2F/).
+
+## Prerequisites for Yubikey Passwordless Authentication
+
+User must have compatible [YubiKey hardware](https://www.yubico.com/products/) to begin with.
+
+## Creating Yubikey U2F Keys
+
+Yubikey U2F keys can be created with [pamu2fcfg](https://developers.yubico.com/pam-u2f/), a module implements PAM over U2F and FIDO2. pamu2fcfg is available in Nixpkgs as pkgs.pamu2fcfg.
+
+Use the following command to create your U2F keys for Ghaf gui-vm .
+```
+$ pamu2fcfg -u ghaf -o pam://gui-vm | sudo tee /etc/u2f_mappings
+```
+
+After running above command, there will be green light blinking on Yubikey and user need to touch it.
+Once executed successfully, now user authentication becomes passwordless. When prompted with password just touch Yubikey.
+More details about mapping file can found here [central-authorization-mapping](https://github.com/Yubico/pam-u2f#central-authorization-mapping)
+
+## Current Implementation
+
+Since passwordless authentication is mapped with Yubikey in use and each Yubikey will generate unique key. So its not possible to use generic key.
+Yubikey Passwordless Authentication feature is enabled in debug builds for Ghaf gui-vm only.
+We are using pre-generated public keys which are created using pam2fcfg tool on different machine and it work automatically on guivm after the public key part is included in the built image for specific Yubikey.
+If user want to register their Yubikey hardware then public key need to generated using pamu2fcfg tool and updated in /etc/u2f_mappings file.
+Kindly note keeping plain public key in Ghaf image is not recommended way and is to be used for demo purpose only.
+
```

Comment:
I checked also vendor/product id of my yubikey with
`lsusb | grep -i Yubico`

The ids happened to be the same as in targets/lenovo-x1-carbon.nix but if they were different I suppose also those need to be updated into the target nix file.

Reply:
Yes @leivos-unikie currently we have tested Yubikey with vendorid=0x1050 and productid=0x0407.
In future if any new requirement comes to support different Yubikey, then we need to verify its functionality and the update target nix file accordingly.

Reply:
`lsusb` ids are vendor id and product id. The passthrough is enabled based on those. If they were different, the user is using another product and yes, the USB passthrough to `guivm` must be updated in that scenario. It might be possible to be more generic with the Yubikey reference implementation passthrough to cover some set of HW keys from Yubico but that implementation can be left for later.

Reply:
I addressed this in [32402a8](https://github.com/tiiuae/ghaf/commit/32402a8d166f6400e47e46a3a802f927ef321088) https://github.com/tiiuae/ghaf/pull/359/files#diff-4c362f5486d800f46a0856fd12dddad0cc6d84e96e798bcfb1adb6cbf19db3c2R19

@mikatammi was working on Hardware definition here https://github.com/tiiuae/ghaf/pull/382 so I think we should leave refactoring these USB `vendorId` and `productId`  to future work https://github.com/tiiuae/ghaf/pull/359/files#diff-789892fdf4e4c650d81ce42dbc84586807ced5e73a8d53c06775acf7df2d0876R116-R184
(copy-paste of `vendorId` and `productId` are minor issue that can be addressed later - the bigger work is required if we want to enable passing through different types of `producId`s of Yubikeys)